src/irmin-pack/conf.mli

```diff
@@ -52,6 +61,10 @@ end
 val spec : Irmin.Backend.Conf.Spec.t
 
 type merge_throttle = [ `Block_writes | `Overcommit_memory ] [@@deriving irmin]
+(** What to do when index cache is full.
+
+    - [`Block_write] will block writes until there is space
+    - [`Overcommit_memory] will use more memory temporarily *)
```

Comment:
What is the index cache in this situation? I took a guess at what the variants mean. I need to verify these but welcome insight if anybody already knows.

Reply:
There is some doc in irmin_pack_intf.ml (you may have seen this already tho!). Here "index cache" I think refers to the on-disk log that index stores updates in, before "merging" the updates into the main index store.

Reply:
Thanks! I hadn't seen that yet.

Reply:
Aha, went to the source and found https://github.com/mirage/index/blob/main/src/index.mli#L25-L30. the log is kept both in-memory and on-disk.